### DISCUSSION TITLE
NextUI Input - Clear button not working

### DISCUSSION THREAD

**Discussion by pablojsx** [heroui-inc/heroui#2198] [Help]
This is a Vite project where I installed NextUI, when I click the clear button nothing happens. I've been developing this app for like 3 months and I just noticed this. I have other Next apps where it works but not in Vite.

**Comment by unav4ila8le** [comment]
Same issue here +1

**Comment by ChrisNSki** [comment]
Also seeing this, to add a bit more detail, I've not changed the component to be controlled, so I shouldn't need to handle the clear on my state

**Comment by J4v4Scr1pt** [comment]
+1

**Comment by alphaxek** [comment]
Hi @pablojsx @J4v4Scr1pt @ChrisNSki @unav4ila8le I created a new Vite project with NextUI and Input clear button is working fine on my side.

can you please provide the code or try updating packages.

Vite test:
![vite](https://github.com/nextui-org/nextui/assets/116849110/f4d3e4d2-d9ac-4aff-ae6d-9c4f3aaa0233)

![image](https://github.com/nextui-org/nextui/assets/116849110/df7de0a7-9b1a-4db2-8697-fa27e8df081b)

Code:
```
import React from "react";
import {Input} from "@nextui-org/react";

export default function App() {
  return (
    <Input
      isClearable
      type="email"
      label="Email"
      variant="bordered"
      placeholder="Enter your email"
      defaultValue="junior@nextui.org"
      onClear={() => console.log("input cleared")}
      className="max-w-xs"
    />
  );
}
```

**Reply by pablojsx** [reply]
That would be the search bar from the index: https://www.ez-tone.com/ but I already replaced it with something different. To be honest I haven't tried creating an empty project to test the input but the index is kinda simple, maybe I did something wrong I can't remember because this is from Jan 02

**Reply by alphaxek** [reply]
okay, great, looks like its sorted on your end.
Site (https://www.ez-tone.com/) looks great.

**if above answer helps, can you please mark it as answered, thanks!**

**Reply by pablojsx** [reply]
Thanks man! and now that I remember about it, I think I had more issues with other components, in total 3 but again I've been out of programming for like a month(I mostly work as a music producer)so I can't remember which components specifically

**Reply by alphaxek** [reply]
that's great!
keep working

**Reply by J4v4Scr1pt** [reply]
My memory fails me in this case 😅 and we have no input atm on our site (migration process). But I can try again later today when I'm on the computer again 😊.

**Reply by alphaxek** [reply]
sure👍

**Reply by J4v4Scr1pt** [reply]
Latest version of nextUI 2.2.10 and the clear button in the Input works like a charm ✌

**Reply by unav4ila8le** [reply]
I can confirm that now it works.
However, it seems to not work when you have an "endContent" on the input.

**Reply by wingkwong** [reply]
@unav4ila8le this is an [known issue](https://github.com/nextui-org/nextui/issues/2254). The fix is being reviewed.
